src/lib.rs

```diff
@@ -851,6 +851,76 @@ impl<L, R> fmt::Display for Either<L, R>
     }
 }
 
+#[cfg(feature = "serde")]
+pub mod serde_untagged {
+    //! Untagged serialization/deserialization support.
+    //!
+    //! `Either` uses default, externally-tagged representation.
+    //! However, sometimes it is useful to support several alternative types.
+    //! For example, we may have a field which is generally Map<String, i32>
+    //! but in typical cases Vec<String> would suffice, too.
+    //! ```rust
+    //! #[macro_use]
+    //! extern crate serde;
+    //! // or `use serde::{Serialize, Deserialize};` in newer rust versions.
+    //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    //! use std::collections::HashMap;
+    //! 
+    //! #[derive(Serialize, Deserialize, Debug)]
+    //! #[serde(transparent)]
+    //! struct IntOrString {
+    //!     #[serde(with="either::serde_untagged")]
+    //!     inner: either::Either<Vec<String>, HashMap<String, i32>>
+    //! };
+    //! // serialization
+    //! let data = IntOrString {
+    //!     inner: either::Either::Left(vec!["Hello".to_string()])    
+    //! };
+    //! // notice: no tags are emitted.
+    //! assert_eq!(serde_json::to_string(&data)?, r#"["Hello"]"#);
+    //! // deserialization
+    //! let data: IntOrString = serde_json::from_str(
+    //!     r#"{"a": 0, "b": 14}"#   
+    //! )?;
+    //! println!("found {:?}", data);
+    //! # Ok(()) 
+    //! }
+    //! ```
+
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+    use super::Either;
+
+    #[derive(Serialize, Deserialize)]
+    #[serde(untagged)]
+    enum UEither<L, R> {
```

Comment:
When I was playing with #22, I found that error messages looked nicer when the inner name shadowed the real `Either` -- we don't really want the user to ever see details like `UEither`. Will that work if you just use `super::Either` where the real thing is needed?